demos/adjoint_2d_cylindrical/forward.py

```diff
@@ -136,14 +119,18 @@ def step_func(centre, mag, increasing=True, sharpness=50):
         z,
         T,
         approximation,
-        mu=mu,
         bcs=stokes_bcs,
         nullspace=Z_nullspace,
         transpose_nullspace=Z_nullspace,
-        near_nullspace=Z_near_nullspace,
-        solver_parameters=newton_stokes_solver_parameters,
+        near_nullspace=Z_near_nullspace
     )
 
+    # Overwrite detault solver parameters to use a direct solver for Stokes system.
+    stokes_solver.solver_parameters["mat_type"] = "aij"
+    stokes_solver.solver_parameters["ksp_type"] = "preonly"
+    stokes_solver.solver_parameters["pc_type"] = "lu"
+    stokes_solver.solver_parameters["pc_factor_mat_solver_type"] = "mumps"
+
```

Comment:
In this case, you should use `solver_parameters="direct"` in the call to `StokesSolver`.